### PR TITLE
change null comparison to not empty

### PR DESCRIPTION
**Pull request by imshashank**

### CODE REVIEW

**Comment by imshashank** [comment]
Fix for https://github.com/aws/aws-sdk-php/issues/1209

**Comment by mtdowling** [comment]
Can you add a test case please?

File: src/S3/PostObjectV4.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ public function __construct(
         $credentials   = $this->client->getCredentials()->wait();
         $securityToken = $credentials->getSecurityToken();
 
-        if (null !== $securityToken) {
+        if (!empty($securityToken)) {
```
Comment (jeskew) [inline src/S3/PostObjectV4.php]: Why not just `if ($securityToken = $credentials->getSecurityToken()) {`?

Comment (imshashank) [inline src/S3/PostObjectV4.php]: I am not sure if using `if ($securityToken = $credentials->getSecurityToken()) {` will check for `isEmpty($securityToken)`. Doesn't it only check for truth/false values?

Comment (jeskew) [inline src/S3/PostObjectV4.php]: `if (!empty($variable))` is shorthand for `if (isset($variable) && $variable)`. Both consider the condition met if `$variable` contains a truthy value and fail if it contains falsy one (including an empty string).

File: tests/S3/PostObjectV4Test.php [hunk 2]
```diff
@@ -92,6 +92,50 @@ public function testSignsPostPolicy()
...
+            'credentials' => [
+                'key' => 'AKIAIOSFODNN7EXAMPLE',
+                'secret' => 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
+                'token' => false
+            ],
+        ]);
+        $p = new PostObjectV4(
+            $client,
+            'sigv4examplebucket',
+            array(),
```
Comment (cjyclaire) [inline tests/S3/PostObjectV4Test.php]: Maybe just [] ?

**Comment by jeskew** [comment]
I think it's also worth making the other change mentioned in #1209. The interface for credentials specifies that `getSecurityToken` returns either a string or null, and environmental credentials are returning `false` for that method.

**Comment by imshashank** [comment]
@jeskew Made the changes asked. Thanks for the suggestions.

**Comment by jeskew** [comment]
LGTM :shipit:

**Pull request merged**